### PR TITLE
Added a build script & a new example LED RGB strip

### PR DESCRIPTION
**Pull request by Dave1001**
Hi there
There are two commits here (sorry I haven't taken the time to figure out how to split them apart to create two pull requests in Git - I'll need to study how Git works!):

1) firstly there is a little bash script to handle setting environment variables and building different examples reliably. I hope you find its useful / this is an updated version from pull request #54 (the options are now improved - works well for normal builds). Files updated in this version are build.sh (new file) and readme.md (updated the instructions)

2) secondly there is an example for a WS2182B LED strip - this is one of the most common RGB led "Neopixel" strips (and I have a bunch lying around!). The example adds it as an accessory with RGB color

Dave

### CODE REVIEW

**Comment by maximkulkin** [comment]
Overall, I'm OK with led_strip example (although it is very much like fireplace example), but the build script IMO is too naive to be useful. E.g. it expects SDK to be in home directory with no means to override it (and if you make it overridable, well, you will have a new variable to worry about), it does not work well for rebuild case (because it will ask you same questions again and if you will answer them differently it might yield binary incompatible components).

It feels like most people who have problems with building examples do not read instructions carefully.

I personally do it like this:
1. Install autoenv:
    ```
    brew install autoenv
2. In my project directory create `.env` file:
    ```shell
    export SDK_PATH=~/Projects/eps8266/esp-open-rtos
    export ESPPORT=/dev/tty.SLAB_USBtoUART
    export FLASH_SIZE=32
    export HOMEKIT_DEBUG=1
    ```
Autoenv extension modifies your shell to source .env file whenever you enter directory with that file or any subdirectory.

Then, you do not actually need to do `make -C <dir> all`, you can just do `cd <dir>` and then you can do simple commands like `make all`, `make test` or `make monitor`.

File: examples/led_strip/led_strip.c [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+#include <homekit/homekit.h>
+#include <homekit/characteristics.h>
+#include "wifi.h"
+#include "ws2812_i2s/ws2812_i2s.h"
+
+#define LED_ON	0				// this is the value to write to GPIO for led on (GPIO Low)
+#define LED_OFF 1				// this is the value to write to GPIO for led off (GPIO High)
+#define LED_INBUILT_GPIO 2		// this is the onboard LED used to show on/off only
+#define LED_STRIP_QUANTITY 16	// this is the number of WS2812B leds on the strip
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: Sure thing - will replace all tabs with spaces (interestingly I just found out about the "untabify" command in the IDE I use)

File: examples/led_strip/led_strip.c [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+#include <FreeRTOS.h>
+#include <task.h>
+#include <math.h>  //requires LIBS ?= hal m to be added to Makefile
+
+#include <homekit/homekit.h>
+#include <homekit/characteristics.h>
+#include "wifi.h"
+#include "ws2812_i2s/ws2812_i2s.h"
+
+#define LED_ON	0				// this is the value to write to GPIO for led on (GPIO Low)
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+void led_hue_set(homekit_value_t value) {
+	if (value.format != homekit_format_float) {
+		// printf("Invalid hue-value format: %d\n", value.format);
+		return;
+	}
+	led_hue = value.float_value;
+	ledStringSet();
+}
+
+homekit_value_t led_sat_get() {
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <espressif/esp_wifi.h>
+#include <espressif/esp_sta.h>
+#include <esp/uart.h>
+#include <esp8266.h>
+#include <FreeRTOS.h>
+#include <task.h>
+#include <math.h>  //requires LIBS ?= hal m to be added to Makefile
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - removed the comment (its already in the makefile)

File: examples/led_strip/led_strip.c [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+		g = LED_RGB_SCALE * i / 3 * (1 - s);
+	}
+
+	rgb->red = (uint8_t) r;
+	rgb->green = (uint8_t) g;
+	rgb->blue = (uint8_t) b;
+	rgb->white = (uint8_t) 0;	// white channel is not used
+}
+
+void ledStringFill(ws2812_pixel_t rgb) {
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+		return;
+	}
+	led_saturation = value.float_value;
+	ledStringSet();
+}
+
+homekit_characteristic_t name = HOMEKIT_CHARACTERISTIC_(NAME, "Sample LED Strip");
+
+homekit_accessory_t *accessories[] = {
+	HOMEKIT_ACCESSORY(.id = 1,.category = homekit_accessory_category_lightbulb,.services = (homekit_service_t*[]) {
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+	// initialise the LED strip
+	ws2812_i2s_init(LED_STRIP_QUANTITY, PIXEL_RGB);
+
+	// set the initial state
+	ledStringSet();
+}
+
+void led_identify_task(void *_args) {
+
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+ws2812_pixel_t pixels[LED_STRIP_QUANTITY];
+
+//http://blog.saikoled.com/post/44677718712/how-to-convert-from-hsi-to-rgb-white
+static void hsi2rgb(float h, float s, float i, ws2812_pixel_t* rgb) {
+	int r, g, b;
+
+	while (h < 0) { h += 360.0F; };		// cycle h around to 0-360 degrees
+	while (h >= 360) { h -= 360.0F; };
+	h = 3.14159F*h / 180.0F;			// convert to radians.
+	s /= 100.0F;						// from percentage to ratio
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - done

File: examples/led_strip/led_strip.c [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+#include "ws2812_i2s/ws2812_i2s.h"
+
+#define LED_ON	0				// this is the value to write to GPIO for led on (GPIO Low)
+#define LED_OFF 1				// this is the value to write to GPIO for led off (GPIO High)
+#define LED_INBUILT_GPIO 2		// this is the onboard LED used to show on/off only
+#define LED_STRIP_QUANTITY 16	// this is the number of WS2812B leds on the strip
+#define LED_RGB_SCALE 255		// this is the scaling factor used for color conversion
+
+// Global constants
+const ws2812_pixel_t COLOR_WHITE = { { 255, 255, 255, 0 } };
```
[the remark this answers is not in the document]
Comment (Dave1001) [reply examples/led_strip/led_strip.c]: OK - removed

**Comment by Dave1001** [comment]
Thanks Maxim - using autoenv is exactly what I was looking for! 

Regarding changes requested:
1) I've removed the bash script (and agree it was clunky). autoenv is better
2) have made the rest of the changes requested. Thanks for your time on this - they should be ready to do a final review and pull

**Pull request merged**